### PR TITLE
fix: wrap dynamic entry functions (#255)

### PR DESCRIPTION
**Pull request by crlfe**
Maintainer has objections on the issue, so am tossing this up to make sure we're talking about the same thing.

This PR contains a:

- [x] **bugfix**
- [ ] new **feature**
- [ ] **code refactor**
- [x] **test update** 
- [ ] **typo fix**
- [ ] **metadata update**

### Motivation / Use-Case

Limitations in webpack-hot-client require that the dynamic entry function always return an array or object with array values. Extending the current mangling means that webpack-serve will work as well as webpack-cli for these cases.

### Breaking Changes

Should not be any changes for non-function entry values.

### Additional Info

webpack documentation allows the dynamic entry function to return a PromiseLike object. I have no need of that, so would prefer to ignore the extra branch and testing (believe the closure would check for a PromiseLike and `.then(entryFixValue)`).

### CODE REVIEW

File: lib/config.js [hunk 1]
```diff
@@ -5,24 +5,37 @@ const isPlainObject = require('is-plain-obj');
...
     result.entry = ['./src'];
-  } else if (isPlainObject(result.entry)) {
-    for (const key of Object.keys(result.entry)) {
-      const entry = result.entry[key];
-
-      if (!Array.isArray(entry)) {
-        result.entry[key] = [entry];
-      }
-    }
+  } else if (result.entry instanceof Function) {
```
Comment (shellscape) [inline lib/config.js]: There's no need for a separate function to be broken out of `toArray`. Please restore the original function signature, add your function handling block, and simply make a call to itself.

File: test/config.test.js [hunk 2]
```diff
@@ -24,6 +24,22 @@ describe('config', () => {
     expect(toArray(config)).toMatchSnapshot();
   });
 
+  test('toArray: String Function', () => {
```
Comment (shellscape) [inline test/config.test.js]: I don't see a test here for handling `thenables`. If `Promises/thenables` aren't supported by this change then the README should be updated in this PR as well.

File: lib/config.js (hunk 1 above)
Comment (crlfe) [reply lib/config.js]: Not clear what you mean by "function signature". The behavior of toArray should not have changed (note the else on line 38).

In any case, I've dropped these tools, so won't waste any more of my time.

**Comment by crlfe** [comment]
Abandoning, since I'm no longer using these tools.

**Pull request closed without merging**